Source/TestingServices/SchedulingStrategies/Liveness/CycleDetectionStrategy.cs

```diff
@@ -640,23 +664,23 @@ private void EscapeUnfairCycle()
         /// </summary>
         private void DebugPrintScheduleTrace()
         {
-            if (Configuration.EnableDebugging)
+            if (Configuration.EnableDebugging && Debug.IsEnabled)
             {
                 Debug.WriteLine("<LivenessDebug> ------------ SCHEDULE ------------.");
 
                 foreach (var step in ScheduleTrace)
                 {
                     if (step.Type == ScheduleStepType.SchedulingChoice)
                     {
-                        Debug.WriteLine($"{step.Index} :: {step.Type} :: {step.ScheduledMachineId} :: {StateCache[step].Fingerprint}");
```

Comment:
These were the only three uses of StateCache[...] I could find, so I removed it.